internal/core/app_status_report.go

```diff
@@ -323,16 +323,16 @@ func (op *statusReportOperation) Do(
 		HealthMessage: report.HealthMessage,
 	}
 
-	// Populate resource health with health in plugin compiled report
-	resourcesHealth := make([]*pb.StatusReport_Health, len(report.Resources))
-	for i, r := range report.Resources {
-		resourcesHealth[i] = &pb.StatusReport_Health{
-			HealthStatus:  r.Health.String(),
-			HealthMessage: r.HealthMessage,
-			Name:          r.Name,
+	// Add our resources
+	var resources []*pb.StatusReport_Resource
+	for _, sdkResource := range report.Resources {
+		var resource pb.StatusReport_Resource
+		if err := mapstructure.Decode(sdkResource, &resource); err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to decode status report resource named %q: %s", sdkResource.Name, err)
 		}
+		resources = append(resources, &resource)
 	}
-	realMsg.ResourcesHealth = resourcesHealth
```

Comment:
I've checked to make sure we aren't using this ResourcesHealth anywhere - it's ok to not populate it.